wallet/wallet-cli-lib/src/lib.rs

```diff
@@ -100,30 +102,30 @@ pub async fn run(
         .await
         .map_err(WalletCliError::RpcError)?;
 
-    let mut controller_opt = None;
+    let controller_opt = None;
+
+    let (event_tx, event_rx) = mpsc::unbounded_channel();
 
     if let Some(wallet_path) = wallet_file {
-        commands::handle_wallet_command(
-            &chain_config,
-            &rpc_client,
-            &mut controller_opt,
-            commands::WalletCommand::OpenWallet { wallet_path },
-        )
-        .await?;
+        let (res_tx, _res_rx) = tokio::sync::oneshot::channel();
+        event_tx
+            .send(Event::HandleCommand {
+                command: WalletCommand::OpenWallet { wallet_path },
+                res_tx,
+            })
+            .expect("should not fail");
```

Comment:
`_res_rx` should be used here, to make sure that the `wallet_file` command line argument handling errors are correctly detected.

Reply:
I put them in a vector and passed them to the `repl_handle` to process/print/error the responses

Reply:
Sorry, but I don't think that's enough, the app should exit early with an error code (like it worked before).

Reply:
Fixed the early exit